### PR TITLE
Implementing Question Visibility

### PR DESCRIPTION
**Pull request by maximusunc**

### CODE REVIEW

File: src/components/shared/QuestionHeader.jsx [hunk 1]
```diff
@@ -43,34 +43,6 @@ class QuestionHeader extends React.Component {
   componentDidMount() {
     this.syncPropsAndState(this.props);
   }
-  shouldComponentUpdate(newProps, newState) {
```
Comment (patrickkwang) [inline src/components/shared/QuestionHeader.jsx:46]: Why did you remove this?

Comment (maximusunc) [reply src/components/shared/QuestionHeader.jsx:46]: I took it out and nothing about the application changed. The component was not rerendering more than necessary.
